Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Plans.hs

```diff
@@ -0,0 +1,169 @@
+          return $ buildSubscriptionResponse (Just order)
+    MANUAL -> do
+      case driverPlan of
+        Just dPlan -> do
+          when (maybe False checkMandateStatus (dPlan.mandateStatus)) (throwError $ InvalidMandateStatus driverId.getId)
+          runNoTransaction $ QDPlan.updatePlanIdByDriverId driverId planId
+          return $ buildSubscriptionResponse Nothing
+        Nothing -> do
+          newDriverPlan <- mkDriverPlanEntry planId driverId paymentMode INACTIVE_PLAN now 0
+          runNoTransaction $ QDPlan.create newDriverPlan
+          return $ buildSubscriptionResponse Nothing
+
+select ::
+  (Id DP.Person, Id DM.Merchant) ->
+  Id PlanDetails ->
+  PaymentMode ->
+  Flow AckResponse
+select (driverId, _) planId paymentMode = do
+  driverInfo <- runInReplica $ QDInfo.findById (cast driverId) >>= fromMaybeM DriverInfoNotFound
+  currentDriverPlan <- runInReplica $ QDPlan.findByDriverId driverId >>= fromMaybeM (NoCurrentPlanForDriver driverId.getId)
+  when (currentDriverPlan.planStatus == INACTIVE_PLAN) (throwError $ InvalidPlanStatus driverId.getId)
+  case driverInfo.planPaymentMode of
+    MANUAL -> do
+      when (paymentMode /= MANUAL) (throwError $ InvalidMandateStatus driverId.getId)
+      runNoTransaction $ QDPlan.updatePlanIdByDriverId driverId planId
+      return Ack
+    AUTOPAY -> do
+      when (paymentMode /= AUTOPAY) (throwError $ InvalidMandateStatus driverId.getId)
+      unless (maybe False checkMandateStatus (currentDriverPlan.mandateStatus)) (throwError $ InvalidMandateStatus driverId.getId)
+      runNoTransaction $ QDPlan.updatePlanIdByDriverId driverId planId
+      return Ack
+
+upgrade ::
+  (Id DP.Person, Id DM.Merchant) ->
+  Flow Payment.CreateOrderResp
+upgrade (driverId, merchantId) = do
+  currentDriverPlan <- runInReplica $ QDPlan.findByDriverId driverId >>= fromMaybeM (NoCurrentPlanForDriver driverId.getId)
+  driverInfo <- runInReplica $ QDInfo.findById (cast driverId) >>= fromMaybeM DriverInfoNotFound
+  when (driverInfo.planPaymentMode == AUTOPAY) (throwError InvalidPaymentMode)
+  unless (currentDriverPlan.planStatus == ACTIVE_PLAN) (throwError $ InvalidPlanStatus driverId.getId)
+  when (maybe True checkMandateStatus (currentDriverPlan.mandateStatus)) (throwError $ InvalidMandateStatus driverId.getId)
+  runNoTransaction $ QDPlan.updatePaymentModeAndStatusByDriverId driverId AUTOPAY INACTIVE_PLAN
+  driverRegisterationInvoice <- mkDriverFeeForMandate 1 driverId --- amount is rs. 1 needs to be verified ---
+  createMandateOrder (driverId, merchantId) driverRegisterationInvoice.id 1.0
+
+checkMandateStatus :: MandateStatus -> Bool
+checkMandateStatus mandateStatus = mandateStatus `elem` [ACTIVE, PAUSED, CREATED]
+
+mkDriverFeeForMandate ::
+  ( MonadFlow m
+  ) =>
+  Money ->
+  Id DP.Person ->
+  m DF.DriverFee
+mkDriverFeeForMandate amount driverId = do
+  id <- generateGUID
+  shortId <- generateShortId
+  now <- getCurrentTime
+  return $
+    DF.DriverFee
+      { payBy = now,
```

Comment:
We generally unsubscribe the driver if payment is not made by this time. Why is it `now`?